pkg/kstatus/watcher/watcher.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package watcher
+
+import (
+	"context"
+
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling/event"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// StatusWatcher watches a set of objects for status updates.
+type StatusWatcher interface {
+
+	// Watch a set of objects for status updates.
+	// Watching should stop if the context is cancelled.
+	// Events should only be sent for the specified objects.
+	// The event channel should be closed when the watching stops.
+	Watch(context.Context, object.ObjMetadataSet, Options) <-chan event.Event
+}
+
+// Options can be provided when creating a new StatusWatcher to customize the
+// behavior.
+type Options struct {
+	// RESTScopeStrategy specifies which strategy to use when listing and
+	// watching resources. By default, the strategy is selected automatically.
+	RESTScopeStrategy RESTScopeStrategy
+}
+
+//go:generate stringer -type=RESTScopeStrategy -linecomment
+type RESTScopeStrategy int
```

Comment:
nit: This can probably be `byte` =)

Reply:
I've never seen a go enum use byte. Int is more extensible. But perhaps you're suggesting that this particular enum will never be extended?